src/steps/profile/index.test.ts

```diff
@@ -26,18 +26,16 @@ describe('#fetchProfiles', () => {
             hostname: false,
           },
         },
-        recordFailedRequests: false,
+        recordFailedRequests: true,
       },
     });
 
-    const context = createMockStepExecutionContext({
-      instanceConfig: integrationConfig,
-    });
-    await fetchProfiles(context);
+    const stepTestConfig = getStepTestConfigForStep(Steps.PROFILES);
+    const stepResults = await executeStepWithDependencies(stepTestConfig);
```

Comment:
Love the pattern update! Thank you! 💯 

What do you think about also adopting the `.toMatchStepMetadata()` jest matcher? Are there any issues when using it, or anything we can update in the SDK to make adopting it easier?

```suggestion
    const stepResults = await executeStepWithDependencies(stepTestConfig);
    expect(stepResult).toMatchStepMetadata(stepTestConfig);
```

Reply:
I tried to do this but failed, we could meet about it. Basically I didn't find a way to `disableClassMatch` using that matcher.
I know AWS uses it as a property in the entity, but that does not work directly in this integration because the entities are well-typed.